src/Mobs/Silverfish.cpp

```diff
@@ -0,0 +1,66 @@
+
+#include "Globals.h"
+
+#include "Silverfish.h"
+
+#include "../World.h"
+#include "../Chunk.h"
+#include "../Blocks/BlockHandler.h"
+#include "../Blocks/BlockInfested.h"
+#include "../Entities/Player.h"
+
+bool cSilverfish::DoTakeDamage(TakeDamageInfo &a_TDI)
+{
+	bool SuperResult = Super::DoTakeDamage(a_TDI);
+	// Todo: stop this if /gamerule mobGriefing is set to false
+
+	// If the entity didn't take andy damage
+	if (!SuperResult)
+	{
+		return SuperResult;
+	}
+
+	// Entity does receive lethal damage or Attacker doesn't exist
+	if ((m_Health < a_TDI.FinalDamage) ||
+		((a_TDI.Attacker == nullptr) && (a_TDI.DamageType != dtPoison) && (a_TDI.DamageType != dtPotionOfHarming)))
+	{
+		return SuperResult;
+	}
+
+	// If attacker is player or splash potion
+	bool ShouldSpawn = false;
+	if ((a_TDI.DamageType == dtPoison) || (a_TDI.DamageType == dtPotionOfHarming))
+	{
+		ShouldSpawn = true;
+	}
+	else
+	{
+		ShouldSpawn |= a_TDI.Attacker->IsPlayer();
+	}
```

Comment:
NIT: This is a bit verbose.
```suggestion
	bool ShouldSpawn = (
		(a_TDI.DamageType == dtPoison) || (a_TDI.DamageType == dtPotionOfHarming) ||
		a_TDI.Attacker->IsPlayer()
	);
```